src/index.ts

```diff
@@ -8,56 +8,105 @@ import * as formatter from "./helpers/formatter";
 
 class MongooseTsgen extends Command {
   static description =
-    'Generate an index.d.ts file containing Mongoose Schema interfaces. The specified root path ("." by default) will be searched recursively for a `models` folder, where your Mongoose Schemas should be exported from.';
+    'Generate a Typescript file containing Mongoose Schema interfaces. The specified root path ("." by default) will be searched recursively for a `models` folder, where your Mongoose Schemas should be exported from.';
 
   static flags = {
     help: flags.help({ char: "h" }),
     output: flags.string({
       char: "o",
-      default: "./src/types/mongoose",
-      description: "path of output index.d.ts file"
+      description:
+        "Path of output file containing generated interfaces. If a folder path is passed, the generator will default to creating an `mongoose.gen.ts` file in the specified folder. Defaults to ./src/interfaces."
     }),
     "dry-run": flags.boolean({
       char: "d",
-      default: false,
-      description: "print output rather than writing to file"
+      description: "Print output rather than writing to file."
     }),
     "no-func-types": flags.boolean({
-      default: false,
-      description: "disable using TS compiler API for method, static and query typings"
+      description: "Disable using TS compiler API for method, static and query typings."
     }),
     "no-format": flags.boolean({
-      default: false,
-      description: "disable formatting generated files with prettier and fixing with eslint"
+      description: "Disable formatting generated files with prettier and fixing with eslint."
     }),
     js: flags.boolean({
       char: "j",
-      default: false,
-      description: "search for Mongoose schemas in Javascript files rather than in Typescript files"
+      description:
+        "Search for Mongoose schemas in Javascript files rather than in Typescript files."
     }),
     project: flags.string({
       char: "p",
-      default: "./",
-      description: "path of tsconfig.json or its root folder"
+      description: `Path of tsconfig.json or its root folder. Defaults to "./"`
+    }),
+    imports: flags.string({
+      char: "i",
+      description: "Custom import statements to ensure your function types are all defined",
+      multiple: true
+    }),
+    config: flags.string({
+      char: "c",
+      description:
+        "Path of mtgen.config.json or its root folder. CLI flag options will take precendence over settings in mtgen.config.json"
+    }),
+    module: flags.boolean({
+      description: `generate interfaces in a "declare module 'mongoose'" block`
```

Comment:
```suggestion
    augment: flags.boolean({
      description: `augment generated interfaces into the 'mongoose' module`
```
I would call this flag `augment` instead of `module`, since that more clearly describes what the flag does.

It might also be useful to link to [the documentation](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation) somehow, because based on my experiences over at DefinitelyTyped a lot of people don't know what module augmentation is.